src/scenic/core/vectors.py

```diff
@@ -4,427 +4,659 @@
+
+    @property
+    def w(self) -> float:
+        return self.q[3]
+
+    @property
+    def x(self) -> float:
+        return self.q[0]
+
+    @property
+    def y(self) -> float:  
+        return self.q[1]
+
+    @property
+    def z(self) -> float: 
+        return self.q[2]
+
+    @property
+    def yaw(self) -> float:
+        """Yaw in the global coordinate system."""
+        return self.eulerAngles[0]
+
+    @property
+    def pitch(self) -> float:
+        """Pitch in the global coordinate system."""
+        return self.eulerAngles[1]
+
+    @property
+    def roll(self) -> float:
+        """Roll in the global coordinate system."""
+        return self.eulerAngles[2]
+
+    @staticmethod
+    def _coerce(thing) -> Orientation:
+        if isinstance(thing, Orientation):
+            return thing
+        elif hasattr(thing, 'toOrientation'):
+            return thing.toOrientation()
+        elif isinstance(thing, Vector):
+            return Orientation.fromEuler(*thing)
+        elif isinstance(thing, (tuple, list)):
+            if len(thing) != 3:
+                raise CoercionFailure("Cannot coerce a tuple/list of length not 3 to an orientation")
+            return Orientation.fromEuler(*thing)
+        elif canCoerceType(type(thing), float):
+            return Orientation.fromEuler(coerceToFloat(thing), 0, 0)
+        else:
+            raise CoercionFailure
+
+    @staticmethod
+    def _canCoerceType(ty):
+        if issubclass(ty, (tuple, list, Vector)):
+            return True
+        return canCoerceType(ty, float) or issubclass(ty, Orientation) or hasattr(ty, 'toOrientation')
+
+    @cached_property
+    def eulerAngles(self) -> typing.Tuple[float, float, float]:
+        """Global intrinsic Euler angles yaw, pitch, roll."""
+        # Wrapped to catch gimbal lock warning.
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            angles = self.r.as_euler('ZXY', degrees=False)
+        return angles
+
+    def getRotation(self):
+        return self.r
+
+    @cached_property
+    def inverse(self) -> Orientation:
+        return Orientation(self.r.inv())
+
+    @distributionMethod
+    def __mul__(self, other) -> Orientation:
+        if type(other) is not Orientation:
+            return NotImplemented
+        return Orientation(self.r * other.r)
+    
+    @distributionMethod
+    def __add__(self, other) -> Orientation:
+        if isinstance(other, (float, int)):
+            other = Orientation.fromEuler(other, 0, 0)
+        elif type(other) is not Orientation:
+            return NotImplemented
+        return other * self
+
+    @distributionMethod
+    def __radd__(self, other) -> Orientation:
+        if isinstance(other, (float, int)):
+            other = Orientation.fromEuler(other, 0, 0)
+        elif type(other) is not Orientation:
+            return NotImplemented
+        return self * other
+
+    def __getitem__(self, index) -> float:
```

Comment:
I don't think this is used anywhere, so I would remove it: people might expect to get the Euler angles for example, given that `(X, Y, Z)` can be used to define an orientation. It's better to explicitly ask for the quaternion.